File: questao12/java/Produto.java
package questao12.java;




public class Produto {
    private String nome;
    private double preco;

    // Construtor
    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    // Método para somar os preços de dois produtos
    public double somarPrecos(Produto outroProduto) {
        return this.preco + outroProduto.preco;
    }

    // Método para exibir os detalhes do produto
    @Override
    public String toString() {
        return nome + " - Preço: " + preco;
    }

    public static void main(String[] args) {
        // Criando dois produtos
        Produto produto1 = new Produto("Produto A", 50.0);
        Produto produto2 = new Produto("Produto B", 30.0);

        // Somando os preços dos dois produtos
        double total = produto1.somarPrecos(produto2);

        System.out.println("Soma dos preços: " + total);
    }
}

File: questao12/go/produto.go
package main

import "fmt"

// Struct Produto
type Produto struct {
    nome  string
    preco float64
}

// Método para somar os preços de dois produtos
func (p Produto) SomarPrecos(outro Produto) float64 {
    return p.preco + outro.preco
}

func main() {
    // Criando dois produtos
    produto1 := Produto{nome: "Produto A", preco: 50.0}
    produto2 := Produto{nome: "Produto B", preco: 30.0}

    // Somando os preços dos dois produtos
    total := produto1.SomarPrecos(produto2)

    fmt.Printf("Soma dos preços: %.2f\n", total)
}
